lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -161,6 +161,19 @@ def initialize
       @prev_time = Time.now
     end
 
+    def periodically_recreate_k8s_client
+      token_read_period = 10
+      while true
+        log.debug "Sleeping for #{token_read_period}"
+        sleep(token_read_period)
```

Comment:
This seems like it will pause all processing.  I believe a better implementation would be to follow the pattern of https://github.com/ManageIQ/kubeclient/pull/532/files

Reply:
@jcantrill 
I'm not really skilled in threads and in ruby, how that having a sleep in a thread (not thread.sleep) pauses all processing ?

Having read about "||=" statement which I didn't know, `@faraday_client ||= create_faraday_client` seems to be equivalent to `@faraday_client || @faraday_client = create_faraday_client` not `@faraday_client = @faraday_client || create_faraday_client` which means `@faraday_client` needs to be set to nil/false to actually be recreated. Have I been misguided ?

Reply:
I don't believe there is any reason to add threading to monitor the client.  It's expiration should happen infrequently enough (e.g. 60 min) that introducing logic to recreate on failure should be sufficient.  I believe the pattern established in https://github.com/ManageIQ/kubeclient/pull/532 is sufficient

Reply:
Actually the pattern is not sufficient when `watch` is false as no metadata will be retrieved.
BTW the periodic approach is the one used in kubernetes client-go to refresh token.